jsonnet/kube-prometheus/prometheus/prometheus.libsonnet

```diff
@@ -18,60 +18,67 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     },
 
     prometheus+:: {
-      name: 'k8s',
-      replicas: 2,
       rules: {},
-      renderedRules: {},
-      namespaces: ['default', 'kube-system', $._config.namespace],
     },
   },
 
   prometheus+:: {
+    local p = self,
+
+    name:: 'k8s',
+    namespace:: $._config.namespace,
+    roleBindingNamespaces:: ['default', 'kube-system', $._config.namespace],
+    replicas:: 2,
+    prometheusRules:: $._config.prometheus.rules,
+    alertmanagerName:: $.alertmanager.service.metadata.name,
+
     serviceAccount:
       local serviceAccount = k.core.v1.serviceAccount;
 
-      serviceAccount.new('prometheus-' + $._config.prometheus.name) +
-      serviceAccount.mixin.metadata.withNamespace($._config.namespace),
+      serviceAccount.new('prometheus-' + p.name) +
+      serviceAccount.mixin.metadata.withNamespace(p.namespace),
     service:
       local service = k.core.v1.service;
       local servicePort = k.core.v1.service.mixin.spec.portsType;
 
       local prometheusPort = servicePort.newNamed('web', 9090, 'web');
 
-      service.new('prometheus-' + $._config.prometheus.name, { app: 'prometheus', prometheus: $._config.prometheus.name }, prometheusPort) +
+      service.new('prometheus-' + p.name, { app: 'prometheus', prometheus: p.name }, prometheusPort) +
       service.mixin.spec.withSessionAffinity('ClientIP') +
-      service.mixin.metadata.withNamespace($._config.namespace) +
-      service.mixin.metadata.withLabels({ prometheus: $._config.prometheus.name }),
-    [if $._config.prometheus.rules != null && $._config.prometheus.rules != {} then 'rules']:
+      service.mixin.metadata.withNamespace(p.namespace) +
+      service.mixin.metadata.withLabels({ prometheus: p.name }),
+
+    rules:
```

Comment:
During a brainstom @metalmatze and me came to the conclusion that conditionally rendering that object adds unnecessary complexity. Instead we suggest to proactively hide this object in consumers if it is unneeded.